classes/fields/avatar.php

```diff
@@ -132,32 +254,72 @@ public function get_avatar( $avatar, $id_or_email, $size, $default = '', $alt =
 				}
 
 				if ( ! empty( $avatar_field ) ) {
-					$user_avatar = get_user_meta( $user_id, $avatar_field . '.ID', true );
+					$avatar_id = get_user_meta( $user_id, $avatar_field . '.ID', true );
 
-					if ( ! empty( $user_avatar ) ) {
-						$attributes = array(
-							'alt'   => '',
-							'class' => 'avatar avatar-' . $size . ' photo',
-						);
+					pods_cache_set( $user_id . '-id', $avatar_id, 'pods_avatars' );
+				}//end if
+			}//end if
+		}//end if
 
-						if ( ! empty( $alt ) ) {
-							$attributes['alt'] = $alt;
-						}
+		return (int) $avatar_id;
+	}
 
-						$user_avatar = pods_image( $user_avatar, array( $size, $size ), 0, $attributes );
 
-						if ( ! empty( $user_avatar ) ) {
-							$avatar = $user_avatar;
+	/**
+	 * Get the avatar user ID based on parameter provided.
+	 *
+	 * @since 2.7.22
+	 *
+	 * @param mixed $id_or_email The Gravatar to retrieve. Accepts a user ID, Gravatar MD5 hash,
+	 *                           user email, WP_User object, WP_Post object, or WP_Comment object.
+	 *
+	 * @return int
+	 */
+	public function get_avatar_user_id( $id_or_email ) {
+		$user_id = 0;
 
-							pods_cache_set( $user_id . '-' . $size, $avatar, 'pods_avatars' );
-						}
-					}
-				}//end if
-			}//end if
-		}//end if
+		if ( is_numeric( $id_or_email ) && 0 < $id_or_email ) {
+			$user_id = $id_or_email;
+		} elseif ( is_object( $id_or_email ) && isset( $id_or_email->user_id ) && 0 < $id_or_email->user_id ) {
+			$user_id = $id_or_email->user_id;
+		} elseif ( is_object( $id_or_email ) && isset( $id_or_email->ID ) && isset( $id_or_email->user_login ) && 0 < $id_or_email->ID ) {
+			$user_id = $id_or_email->ID;
+		} elseif ( ! is_object( $id_or_email ) && false !== strpos( $id_or_email, '@' ) ) {
+			$_user = get_user_by( 'email', $id_or_email );
 
-		return $avatar;
+			if ( ! empty( $_user ) ) {
+				$user_id = $_user->ID;
+			}
+		}
+
+		return (int) $user_id;
+	}
+
+	/**
+	 * Checks if we're not on WordPress admin pages where we shouldn't overwrite.
+	 *
+	 * @since 2.7.22
+	 *
+	 * @return bool
+	 */
+	public function allow_avatar_overwrite() {
+
+		// Don't replace for the Avatars section of the Discussion settings page.
+		if ( is_admin() && ! doing_action( 'admin_bar_menu' ) ) {
+			$current_screen = get_current_screen();
+
+			$screens = array(
```

Comment:
@sc0ttkclark Do you agree on this part?

– User edit page (since this is the actual Gravatar)
– Profile edit page (since this is the actual Gravatar)
– Discussion page (user avatar preview options).

For these locations the Gravatar will not be overwritten since that is WordPress core.

Reply:
Sounds good to me, the edit page could be optional though, maybe someone might want to know what that avatar would be (if using the avatar field)

Reply:
The Pods field would still be there so that is still available in those screens.
IMO that is sufficient :)